### ISSUE TITLE
Trakt authorization via settings

### ISSUE THREAD

**Issue by noahf1** [I-A-C/ExodusReduxRepo#15]
Hey,

in Kodi18 it is not possible to start the trakt authentication via the addon settings. The problem is the duplicate of the entry "trakt.user" which is also used to determine if the trakt option should displayed for the indicator setting.

I fixed it that way: https://github.com/lastship/plugin.video.lastship/commit/68dcaabf0aab3d34c55992cd26f29f28e876a4cd

but now the indicator setting cannot determine if trakt is authorized. Any better ideas?

**Comment by bleaklodge** [comment]
> but now the indicator setting cannot determine if trakt is authorized. Any better ideas?

Yes, do not remove text-type setting from the first settings category.
You can have duplicate setting-ids on different categories, if they are the same settings type. They can also have different attributes.
Also all duplicates are automatically updated if their value changes in one of the categories.

**Comment by host505** [comment]
> but now the indicator setting cannot determine if trakt is authorized. Any better ideas?

Yes, do not remove text-type setting from the first settings category.
You can have duplicate setting-ids on different categories, if they are the same settings type. They can also have different attributes.
Also all duplicates are automatically updated if their value changes in one of the categories.

**Comment by SerpentDrago** [comment]
Could someone make a pull request with the fix for this ? thanks

**Comment by noahf1** [comment]
> > but now the indicator setting cannot determine if trakt is authorized. Any better ideas?
> 
> Yes, do not remove text-type setting from the first settings category.
> You can have duplicate setting-ids on different categories, if they are the same settings type. They can also have different attributes.
> Also all duplicates are automatically updated if their value changes in one of the categories.

I will try is this afternoon and create a PR

**Comment by host505** [comment]
> Yes, do not remove text-type setting from the first settings category.
You can have duplicate setting-ids on different categories, if they are the same settings type. They can also have different attributes.
Also all duplicates are automatically updated if their value changes in one of the categories.

Well, unfortunately that will disappear the value from the duplicate entry (trakt username). Only the first setting-id of the duplicates can display the value.

**Comment by jewbmx** [comment]
Here ya go, dont touch anything besides changing to this...

<setting type="action" label="32306" option="close" action="RunPlugin(plugin://plugin.video.exodusredux/?action=authTrakt)" />

**Comment by jewbmx** [comment]
`<setting type="action" label="32306" option="close" action="RunPlugin(plugin://plugin.video.Scrubs/?action=authTrakt)" />`

tested in my addon on the wonderbox 18 apk

**Comment by jewbmx** [comment]
Nevermind my buddy just sent me his debug log and this is now poppin up for him lol, i really hate kodi 18 with a passion lmao

`14:50:09.185 T:547467891792   ERROR: NEWADDON Unknown Video Info Key "fanart2"
14:50:09.185 T:547467891792   ERROR: NEWADDON Unknown Video Info Key "imdb"
14:50:09.185 T:547467891792   ERROR: NEWADDON Unknown Video Info Key "imdb_id"
14:50:09.185 T:547467891792   ERROR: NEWADDON Unknown Video Info Key "metacache"
14:50:09.185 T:547467891792   ERROR: NEWADDON Unknown Video Info Key "next"
14:50:09.185 T:547467891792   ERROR: NEWADDON Unknown Video Info Key "poster"
14:50:09.185 T:547467891792   ERROR: NEWADDON Unknown Video Info Key "poster3"
14:50:09.185 T:547467891792   ERROR: NEWADDON Unknown Video Info Key "tmdb"
14:50:09.185 T:547467891792   ERROR: NEWADDON Unknown Video Info Key "tmdb_id"
`
